src/php/Command/CommandDependencyProvider.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Command;
+
+use Gacela\Framework\AbstractDependencyProvider;
+use Gacela\Framework\Container\Container;
+use Phel\Compiler\CompilerFacade;
+use Phel\Compiler\CompilerFacadeInterface;
+use Phel\Formatter\FormatterFacade;
+use Phel\Formatter\FormatterFacadeInterface;
+use Phel\Interop\InteropFacade;
+use Phel\Interop\InteropFacadeInterface;
+use Phel\Runtime\RuntimeFacade;
+use Phel\Runtime\RuntimeFacadeInterface;
+
+final class CommandDependencyProvider extends AbstractDependencyProvider
+{
+    public const FACADE_COMPILER = 'FACADE_COMPILER';
+    public const FACADE_FORMATTER = 'FACADE_FORMATTER';
+    public const FACADE_INTEROP = 'FACADE_INTEROP';
+    public const FACADE_RUNTIME = 'FACADE_RUNTIME';
+
+    public function provideModuleDependencies(Container $container): void
+    {
+        $this->addFacadeCompiler($container);
+        $this->addFacadeFormatter($container);
+        $this->addFacadeInterop($container);
+        $this->addFacadeRuntime($container);
+    }
+
+    private function addFacadeCompiler(Container $container): void
+    {
+        $container->set(self::FACADE_COMPILER, function (Container $container): CompilerFacadeInterface {
```

Comment:
Couldn't this class be be simplified:

```php
    private function addToContainer(Container $container, string $name, string $class): void
    {
        $container->set($name, function (Container $container) use ($class): CompilerFacadeInterface {
            return $container->getLocator()->get($class);
        });
    }

    public function provideModuleDependencies(Container $container): void
    {
        $this->addToContainer($container, self::FACADE_COMPILER, CompilerFacade::class);
        // ...
    }
```

Reply:
I see your point, but in your example, it wouldn't be possible to define the return type for the anonymous function which is using the locator to resolve the dependency. Also, the scope of this is pretty straight forward: To see which dependencies are we defining (once per function):
```php
$this->addFacadeCompiler($container);
$this->addFacadeFormatter($container);
$this->addFacadeInterop($container);
$this->addFacadeRuntime($container);
```
And I think it wouldn’t be a problem to leave it the way it is (at least for now).